ui/app/js/controllers/rules_controller.js

```diff
@@ -216,6 +236,37 @@ function($scope, $routeParams, $location, $timeout, Rules, Search, $modal, $rout
     return true;
   };
 
+  $scope.$on('$routeChangeSuccess', function () {
+    $timeout(function () {
+      populateRulesProductsChannels();
```

Comment:
Does this actually do anything? populateRulesProductsChannels has no side effects, and you're not using the return value anywhere.